src/components/inputs/checkbox-input/checkbox.js

```diff
@@ -11,48 +11,50 @@ const Input = styled.input`
   }
 `;
 
-class Checkbox extends React.Component {
-  static displayName = 'Checkbox';
-
-  static propTypes = {
-    id: PropTypes.string,
-    name: PropTypes.string,
-    value: PropTypes.string,
-    isChecked: PropTypes.bool,
-    isIndeterminate: PropTypes.bool,
-    onChange: PropTypes.func.isRequired,
-    isDisabled: PropTypes.bool,
-  };
-
-  componentDidMount() {
-    if (this.props.isIndeterminate) {
-      this.ref.current.indeterminate = true;
+const Checkbox = props => {
+  const ref = React.useRef();
+  React.useEffect(() => {
+    if (props.isIndeterminate) {
+      ref.current.indeterminate = true;
     }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.isIndeterminate !== this.props.isIndeterminate) {
-      this.ref.current.indeterminate = this.props.isIndeterminate;
-    }
-  }
-
-  ref = React.createRef();
-
-  render() {
-    return (
-      <Input
-        css={accessibleHiddenInputStyles}
-        id={this.props.id}
-        name={this.props.name}
-        value={this.props.value}
-        disabled={this.props.isDisabled}
-        checked={this.props.isChecked && !this.props.isIndeterminate}
-        onChange={this.props.onChange}
-        ref={this.ref}
-        {...this.props}
-      />
-    );
-  }
-}
+  }, [props.isIndeterminate]);
+
+  const { onChange } = props;
+
+  const handleChange = React.useCallback(
+    event => !props.isReadOnly && onChange && onChange(event),
```

Comment:
Why use JS? Why isn't setting `readOnly` on the input enough?

https://stackoverflow.com/questions/155291/can-html-checkboxes-be-set-to-readonly

Reply:
Thanks for sharing. I often wanted to ask UX "what's the different line in communication towards the user between readonly and disabled? When use which?"